DataFormats/GEMDigi/interface/GEMAMCStatus.h

```diff
@@ -34,9 +34,13 @@ class GEMAMCStatus {
     amcNum_ = amc.amcNum();
     Errors error{0};
     error.badEC = (amc13->lv1Id() != amc.lv1Id());
-    error.badBC = (amc13->bunchCrossing() != amc.bunchCrossing());
+    // Last BC in AMC13 is different to TCDS, AMC, and VFAT
+    error.badBC = !((amc13->bunchCrossing() == amc.bunchCrossing()) ||
+                    (amc13->bunchCrossing() == 0 && amc.bunchCrossing() == GEMAMC13::lastBC));
     error.badRunType = amc.runType() != 0x1;
-    error.badOC = (uint16_t(amc13->orbitNumber()) != amc.orbitNumber());
+    // Last OC in AMC13 is different to TCDS, AMC, and VFAT
+    error.badOC = !((uint16_t(amc13->orbitNumber()) == amc.orbitNumber()) ||
+                    (amc13->bunchCrossing() == 0 && amc.orbitNumber() + 1 == (uint16_t(amc13->orbitNumber()))));
```

Comment:
Did you check that `amc.orbitNumber() + 1` is still considered an `uint16_t`? If yes, that's fine.
However, would I have wanted to be sure that the same data format is used on both sides, I would have rather written
```suggestion
                    (amc13->bunchCrossing() == 0 && uint32_t(amc.orbitNumber() + 1) == amc13->orbitNumber())));
```
(On a side note: is the different format really needed for the `orbitNumber()` in `GEAMAMC` and `GEMAMC13`, since they aro both accessor for a data member wich is `uint64_t` in fact?)

Reply:
Hi @perrotta, thanks for the feedback.
`amc.orbitNumber() + 1`  works fine
I set the size different since orbitNumber for AMC13 has 32 bits, while only 16 bits in AMC.
I've changed AMC to uint32_t.

Reply:
Hi @perrotta, @jshlee, I'm not sure that converting the AMC orbit counter (`amc.orbitNumber()`) to a `uint32_t` would give the expected behavior.

Both the AMC and AMC13 orbit counters come from hardware. The orbit counter on the AMC is 16-bits, rolling over, while the AMC13 orbit counter is 32-bits, rolling over as well. The situation where `amc.orbitNumber() + 1` overflows in C++ exists, but, as far as I know, the overflow of an unsigned integer is well-defined.

This would give the expected behavior as well since we want to compare the lowest 16-bits of the AMC13 orbit counter to the 16-bits of the AMC orbit counter. On the other hand, comparing the full 32-bits number would not give the expected behavior since the AMC orbit counter is at most `2**16`.

Reply:
@lpetre-ulb, then I should revert back. Thanks for catching this.

Reply:
Hi @perrotta, thank you for your sharing your reasoning. Indeed, I missed the point that `amc.orbitNumber() + 1` was promoted to bigger than a `uint16_t` in that context. I just check that casting it back to `uint16_t` give the correct behavior: `uint16_t(amc.orbitNumber() + 1)`. Thanks again to you and @jshlee for the prompt replies.